code/datums/elements/turf_transparency.dm

```diff
@@ -1,27 +1,28 @@
 
 /datum/element/turf_z_transparency
 	element_flags = ELEMENT_DETACH
-	var/show_bottom_level = FALSE
 
 ///This proc sets up the signals to handle updating viscontents when turfs above/below update. Handle plane and layer here too so that they don't cover other obs/turfs in Dream Maker
-/datum/element/turf_z_transparency/Attach(datum/target, show_bottom_level = TRUE)
+/datum/element/turf_z_transparency/Attach(datum/target)
 	. = ..()
 	if(!isturf(target))
 		return ELEMENT_INCOMPATIBLE
 
 	var/turf/our_turf = target
 
-	src.show_bottom_level = show_bottom_level
-
-	our_turf.plane = OPENSPACE_PLANE
 	our_turf.layer = OPENSPACE_LAYER
+	if(istype(our_turf, /turf/open/openspace))
+		our_turf.plane = OPENSPACE_PLANE
```

Comment:
Do not like this. Why nuke the argument?

Reply:
Several important reasons:

`show_bottom_level` was used both as a argument AND as a proc with the same name. This was a mindfuck and was creating confusion as to what is being used where.  Also the name was misleading for both cases.

So then why didn't I just rename the argument to something else?  Well I did initially, but then it turns out the only time this arg was actually getting used was to check if a turf was `/turf/open/openspace` since I don't ever foresee a new turf that behaves like openspace anytime soon, we don't need to make it into an arg.  Also to back that claim up, it's been 5 years since this code was made and there still isn't any new types of openspace.  It'd be like someone making a different version of `/turf/open/space`.  So hardcoding the check in this case is fine and the arg only gets used in two spots.

Reply:
I meant is_openspace, which while it should maybe be renamed seemed valuable.
Being able to control if a tile should join the openspace plane or not is something I forsee being helpful.
The same can actually be said of show_bottom_level, unfucking multiz space is on my bucket list

Reply:
The `openspace` turf is one of a kind.  It's similar to bedrock blocks for minecraft.  We do not want stuff falling through a z-level that is supposed to be rock bottom.  Make sense?  That's what the typecheck is for.  No other turf is going to inherit this behaviour that I can think of now or ever.  Thus it behaves like bedrock.  If anything, maybe a subclass might be made (I don't see why) and this check would still be valid to work with the new subclass.  

Keep in mind, it's been 5 years this code has been here and nothing has changed.  There is no point in making this a modular piece of code via making it into an argument or a trait.

We do however want multi Z transparency to be modular. (which is what this PR helps to do)

Reply:
I don't see any reason it couldn't be supported. Depending on how profiles work out it's something I'm considering using with #63255, and I'd like to have that option open.

Is it just a cleanliness thing? or is there a deeper issue behind it